libpod/image/image.go

```diff
@@ -1075,3 +1077,62 @@ func (i *Image) Comment(ctx context.Context, manifestType string) (string, error
 	}
 	return ociv1Img.History[0].Comment, nil
 }
+
+// Tree gets dependencies of image
+func (i *Image) Tree(ctx context.Context) (*bytes.Buffer, error) {
+	treeDepth := 1
+	rootImage := i
+
+	//Traverse to the root(base) image
+	parent, err := i.GetParent()
+	if err != nil {
+		return nil, err
+	}
+
+	for parent != nil {
+		rootImage = parent
+		parent, err = parent.GetParent()
+		if err != nil {
+			break
+		}
+		treeDepth++
+	}
+
+	var buffer bytes.Buffer
+	//Print all subtree(peers) to the depth of input image
+	err = printSubTree(&buffer, rootImage, 0, "└─ ", treeDepth)
+
+	return &buffer, err
+}
+
+func printSubTree(buffer *bytes.Buffer, root *Image, rootSize uint64, prefix string, treeDepth int) error {
+	if treeDepth == 0 {
+		return nil
+	}
+	size, err := root.Size(context.Background())
+	if err != nil {
+		return err
+	}
+	//Size of current layer will be current layer size minus upper layer
+	buffer.WriteString(fmt.Sprintf("%s ID: %s  Tags: %v VirtualSize: %s\n",
+		prefix,
+		root.ID()[:12],
+		root.Names(),
+		units.HumanSizeWithPrecision(float64(*size-rootSize), 4)),
+	)
+
+	children, err := root.GetChildren()
```

Comment:
(AFAICS a _single_ `libpod/image.Image.GetChildren` is O((total number of images known locally)^2), so the total runtime of `libpod image tree` would be O(total^3)! That’s a very bad worst case.

Meanwhile, with a good enough data structure all of this data should be collectable in O((total number of images known locally)+(size of output)). But this is not worth worrying about until the logic is decided on, and maybe even then not, until we run into a big enough user to notice.)

Reply:
I see the point. Just use the Functions available, but internally, it is parsing all images every time. This can be avoided.

Reply:
Sure, changing `GetChildren` to be O(N) instead of O(N^2) seems trivially possible. But, first things first.